### PR TITLE
Update Ubuntu install instructions for 2020 fall

### PR DESCRIPTION
**Pull request by joelostblom**
I am making this a draft because I am adding a few comments to the changes, will ping when ready for review.

### CODE REVIEW

File: resources_pages/install_ds_stack_ubuntu.md [hunk 1]
```diff
@@ -14,10 +14,10 @@ These instructions will walk you through installing the required Data Science so
 - uses English as the default language
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md:15]: Is the 4 years requirement pretty stringent? And could maybe even be seen an not very inclusive? Are we doing anything that is more demanding in recent MDS than previous years, or could this req be updated to indicate laptops made since the early 2010s or so are ok?

File: resources_pages/install_ds_stack_ubuntu.md [hunk 2]
```diff
@@ -14,10 +14,10 @@ These instructions will walk you through installing the required Data Science so
...
 - student user has full administrative access to the computer
 
-**Students' whose laptops do not meet the requirements specified above will not be able to receive techinical assistance from the MDS team in troubleshooting installation issues.**
+**Students' whose laptops do not meet the requirements specified above will not be able to receive technical assistance from the MDS team in troubleshooting installation issues.**
 
 ## Table of Contents
-- [Google Chrome browser](#chrome-browswer)
+- [Google Chrome browser](#chrome-browser)
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md:22]: I'm going to test firefox now for what we discussed.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ subtitle: MDS software stack install instructions for Ubuntu
 ---
 These instructions will walk you through installing the required Data Science software stack for the UBC Master of Data Science program. Before starting, ensure that your laptop meets our program requirements:
 
-- runs one of the following operating systems: macOS 10.15.X (Catalina) or later, Ubuntu 20.04 or later, Windows 10 Professional, Enterprise or Education; version 2004 or later. Please note that Windows 10 Home is not sufficient as not all the software required for the program can be installed on that OS.
+- runs one of the following operating systems: macOS 10.15.X (Catalina) or later, Ubuntu 20.04 or later, Windows 10 Professional, Enterprise or Education; version 2004 or later. *Please note that Windows 10 Home is not sufficient as not all the software required for the program can be installed on that OS.*
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md]: Should we recommend that they check the box for "install third party media" when installing ubuntu? It is unclicked by default and includes things like flash and proprietary drivers. I don't know if we *need* these for anything and I believe they can be installed later, but I generally include it because I find it useful and non-harmful.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ subtitle: MDS software stack install instructions for Ubuntu
 ---
 These instructions will walk you through installing the required Data Science software stack for the UBC Master of Data Science program. Before starting, ensure that your laptop meets our program requirements:
 
-- runs one of the following operating systems: macOS 10.15.X (Catalina) or later, Ubuntu 20.04 or later, Windows 10 Professional, Enterprise or Education; version 2004 or later. Please note that Windows 10 Home is not sufficient as not all the software required for the program can be installed on that OS.
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md:8]: Should we remove "or later" from all of these? If we want them to all run the same OS version. Or is that too restrictive? For example the big Mac OS updates you mentioned sounds like a good idea for students not to install if they affect any of our software negatively.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 5]
```diff
@@ -96,65 +78,67 @@ This step is required for
...
 ## Git
 
-We will be using the command line version of Git as well as Git through RStudio and Jupyter lab. There are some new Git commands that we will use that are only available as of Git 2.23 (or newer), thus to get this newest version we will ask you to update your Git even if you have already installed it. 
+We will be using the command line version of Git as well as Git through RStudio and Jupyter lab. Some of the Git commands we will use are only available since Git 2.23, so if you're Git is older than this version, we ask you to update it using the following command:
 
 ```
-sudo apt-add-repository ppa:git-core/ppa
-sudo apt-get update
-sudo apt-get install git
+sudo apt update
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md:91]: I changed to `apt` instead of `apt-get` everywhere since there was a mix before and since `apt` has some userfacing conveniences like a progress bar.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 6]
```diff
@@ -164,68 +148,70 @@ Python 3.7.7
...
 conda config --add channels conda-forge
 
 ```
 
-Thus, to install other pacakges individually, we can now use the following: conda install <package-name>. We will install the key packages needed for the start of our program now:
+To install packages individually, we can now use the following command: `conda install <package-name>`. Let's install the key packages needed for the start of our program:
 
 ```
-conda install --yes \
+conda install \
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md:176]: I removed `--yes` as I think it is good that people get used to checking what will be installed to understand a little what is happening. And they can add `-y` themselves if they find the confirmation really annoying.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 7]
```diff
@@ -247,17 +233,15 @@ For more information about these matters see
 https://www.gnu.org/licenses/.
 ```
 
-> Note: For MDS it is important that you have R 4.0 or higher.
-
 ### RStudio
 
 Chose and download the Ubuntu 18/Debian 10 version of RStudio Desktop from [https://www.rstudio.com/products/rstudio/download/#download](https://www.rstudio.com/products/rstudio/download/#download). Open the file and follow the installer instructions.
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md]: Do we recommend them to use the update mechanisms within RStudio for newer versions?

Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: Or does this somehow also add Rstudio to the system repos so it is updated via `apt`?

On a similar note, `snap` packages are on a different update mechanism than `apt` and automatically updates in the background regularly.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 8]
```diff
@@ -41,11 +41,11 @@ In MDS we will be using many tools that work most reliably on the Google Chrome
...
 
 ## Visual Studio Code
 
-We need a powerfull but lightwight text editor, as well as a full-blown Python IDE for more complex analysis projects, the open-source text editor Visual Studio Code (VS Code) can serve both of these purposes for us. You can download VS Code at [https://code.visualstudio.com/download](https://code.visualstudio.com/download). The getting started instructions are here: [https://code.visualstudio.com/docs/?dv=linux64_deb](https://code.visualstudio.com/docs/?dv=linux64_deb). 
+The open-source text editor Visual Studio Code (VS Code) is both a powerful text editor and a full-blown Python IDE, which we will use for more complex analysis. You can install VS Code either via the [Snap store/Ubuntu software app](https://snapcraft.io/code) or via the downloadable deb-file from the VS code website [https://code.visualstudio.com/download](https://code.visualstudio.com/download). The getting started instructions are here: [https://code.visualstudio.com/docs/?dv=linux64_deb](https://code.visualstudio.com/docs/?dv=linux64_deb).
 
 You can test that VS code is installed and can be opened from Terminal by restarting terminal and typing the following command:
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md:55]: Do we care about making a distinction between 'terminal' and 'shell' this early? Or is it good to just say 'terminal' because that is the name in the ubuntu dash for the app?

File: resources_pages/install_ds_stack_ubuntu.md [hunk 9]
```diff
@@ -318,26 +283,27 @@ Sometimes a kernel loads, but doesn't work as expected. To test whether your ins
...
 ```
 
-The above is all we need to have LaTeX work with R Markdown documents, however to for Jupyter we need to add the TinyTex executables to our `PATH` and install several more packages. 
+The above is all we need to have LaTeX work with R Markdown documents, however to for Jupyter we need to add the TinyTex executables to our `PATH` and install several more packages.
 
 To add the TinyTex executables to our `PATH` we need to edit our `.bashrc` file. The TinyTex executables are usually installed in `$HOME/bin`. Thus, add the lines below to the bottom of your `.bashrc` file (which you can open by `code $HOME/.bashrc`):
 
 ```
 # add TinyTex executables to the path
 export PATH="$HOME/bin:$PATH"
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md]: Is it safe to prepend this directory or should we append it instead? I don't know what is higher risk, unwanted executables in this dir that get unwanted preference over other system packages, or unintentional addition of latex packages to other path dirs that get preference over tinyTex unless it is first.

**Comment by joelostblom** [comment]
@ttimbers Ready for review! I have gone through everything except Chrome and LastPass instructions. ~Also it seems like I can't assign people to review, I might need different permissions on this repo.~ (my bad, I had not accepted your invite to the team)

File: resources_pages/install_ds_stack_ubuntu.md [hunk 10]
```diff
@@ -357,110 +333,73 @@ tlmgr install eurosym \
...
-
-You will use Docker to create reproducible, sharable and shippable computing environments for your analyses. For this you will need a Docker account. You can sign up for a free one [here](https://store.docker.com/signup?next=%2F%3Fref%3Dlogin).
+## PostgreSQL
 
-After signing-up, you also need to install Docker **CE** for Ubuntu. Follow the installation instructions using the "Install using the repository" methods found [here](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository).
+We will be using PostgreSQL as our database management system. To install PostgreSQL type the following command:
 
-Next, follow the Linux post installation steps [here](https://docs.docker.com/engine/install/linux-postinstall/) so that you can run Docker without typing `sudo`. 
+```
+sudo apt install postgresql postgresql-contrib
```
Comment (joelostblom) [inline resources_pages/install_ds_stack_ubuntu.md:378]: postgres 12 is now in the repos so I don't think we need to add the separate key and repo unless there was another reason for that then to get a recent version

**Comment by joelostblom** [comment]
Just confirmed that chrome and lastpass instructions are correct.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 1 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:15]: This is because we have had problems with older laptops, and students push the limits of this. I honestly think laptops made in early 2010's will face two problems:
- not enough RAM and hard disk space
- might not be able to update to the latest OS we require

File: resources_pages/install_ds_stack_ubuntu.md (hunk 2 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:22]: I have emailed LT hub to add you as a course assistant to one of my Canvas course shells so you can test this. This should happen sometime this week hopefully. However, let's not have this PR wait on this. Let's open an issue in this repo to add Firefox pending success using it with Canvas quizzes + LaTeX. We can add it with another PR later.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 3 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Sure, this sounds good to me! I am much less experienced with ubuntu and so happy to defer to you here!

File: resources_pages/install_ds_stack_ubuntu.md (hunk 4 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:8]: Sure, I think that's an even better idea. Let's try this and see how well this works. We will need to email Vanessa to update the MDS website as it also appears on this page and we need to keep these in sync. I will do that and cc you.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 3 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:9]: Can you make this change here and then I can copy and paste this in the email?

File: resources_pages/install_ds_stack_ubuntu.md (hunk 4 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:8]: Can you make this change here and then I can copy and paste this in the email?

File: resources_pages/install_ds_stack_ubuntu.md (hunk 5 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:91]: Perfect! 🎉

File: resources_pages/install_ds_stack_ubuntu.md (hunk 6 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:176]: I don't know that I agree here... I don't think that folks will necessarily know to add `--yes` if we don't put it there. And they might not know to say yes (nor how to) when they are installing things. There are a number of Windows users who are going to try dual boot, these folks know nothing of command line installs, so I'd like to make it easier for them.

Also, when you write installs in a Dockerfile, or other similar applications (e.g., Ansible) you use the `--yes`. So I don't see the problem when we will have to teach this to them anyways...

File: resources_pages/install_ds_stack_ubuntu.md (hunk 7 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: We haven't previously recommended any update mechanism. To be honest, Jenny has pushed the students to use the preview version. I am being a bit more conservative here telling them to install the most recent version. You can add a note if you think its helpful.

I know very little about auto updates of RStudio on any OS. I defer to you here.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 8 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:55]: As we discussed today, let's go with terminal.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 11]
```diff
@@ -164,68 +155,72 @@ Python 3.7.7
...
-To obtain the latest R 4.0 packages, type the following into the terminal:
+R is another programming language that we will be using a lot in the MDS program. We will use R both in Jupyter notebooks and in RStudio.
 
-```
-sudo apt-add-repository 'deb https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/'
-```
+#### R
 
-Next, add the key ID for the CRAN network:
+The version of R available in the default Ubuntu repositories (3.6) is older than the one we use in MDS (4.0). To obtain the latest R 4.0 packages, we need to add a new repository which is maintained directly by the r-project. To do this, first add the key for this repository by typing the following:
```
Comment (ttimbers) [inline resources_pages/install_ds_stack_ubuntu.md]: We might want to use more clear language here 3.6.* and 4.0.*, respectively, or 4.0.0 or higher (more human readable) so that they don't fret about not matching on the patch version number.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 9 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Sure, change it to append instead (assuming it still works, which it should). I wondered that as I was doing it...

~~Also, can you confirm that after these LaTeX install instructions you can create PDFs from Jupyter?~~ I see you do this later.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 10 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:378]: Nope, just my lack of ubuntu knowledge! Please go ahead and fix this.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 12]
```diff
@@ -62,32 +68,15 @@ x64
...
-- all homework submission and grading
-- working collaboratively
-
-## GitHub
-
-In MDS we will use [GitHub.com](https://github.com/) as well as an Enterprise version of GitHub hosted here at UBC, [GitHub.ubc.ca](https://github.ubc.ca). Please follow the set-up instructions for both below.
-
-#### GitHub.com
-If you do not yet have one, sign up for a free account at [GitHub.com](https://github.com/).
+Sign up for a free account at https://github.com/ if you don't have one already.
```
Comment (ttimbers) [inline resources_pages/install_ds_stack_ubuntu.md]: Not sure that https://github.com/ will show up as a hyperlink on the website, you might need to change to [https://github.com/](https://github.com/) or something like that.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 13]
```diff
@@ -164,68 +155,72 @@ Python 3.7.7
...
+`conda` will show you the packages that will be downloaded,
+and you can press enter to proceed with the installation.
+
 > Note: we will use many more packages than those listed above across the MDS program, however we will manage these using virtual environments (which you will learn about in DSCI 521: Platforms for Data Science).
 
-## Jupyter extensions
-We will be using several Jupyter extensions that help us use Juypter notebooks more smoothly with Git & GitHub. To install them, paste the following in the terminal below:
+## JupyterLab extensions
+
+We will be using a couple of JupyterLab git extensions to facilitate using Jupyter notebooks with Git & GitHub. Install them via the following commands:
```
Comment (ttimbers) [inline resources_pages/install_ds_stack_ubuntu.md]: These are not all git extensions technically. Jupytext is kind of like nbconvert on steriods. So I would prefer to not refer to all of these as git extensions.

File: resources_pages/install_ds_stack_ubuntu.md [hunk 14]
```diff
@@ -267,37 +260,19 @@ The `tidyverse` R package (and some others) have external dependencies on Ubuntu
 sudo apt install libcurl4-openssl-dev libssl-dev libxml2-dev
 ```
 
-We will install the key R packages needed for the start of our program now:
+Next, install the key R packages needed for the start of MDS program,
+by typing the following into the R terminal in RStudio:
 
 ```
-install.packages(c('tidyverse', 'tinytex', 'blogdown', 'xaringan', 'renv'))
+install.packages(c('tidyverse', 'tinytex', 'blogdown', 'xaringan', 'renv', 'IRkernel'))
```
Comment (ttimbers) [inline resources_pages/install_ds_stack_ubuntu.md]: Let's remove `tinytex` here and just install it later under the LaTeX section, as I think this logically makes more sense (as they need another command to install it as well).
```suggestion
install.packages(c('tidyverse', 'blogdown', 'xaringan', 'renv', 'IRkernel'))
```

**Review by ttimbers — CHANGES_REQUESTED** [review]
There were a few other comments I made before officially starting this review (sorry), so please read through all my comments that I posted tonight in addition to these. Then ping me when done and I will accept the PR.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 4 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md:8]: Done

File: resources_pages/install_ds_stack_ubuntu.md (hunk 1 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md:15]: For the first point, I didn't mean that the extended age limit would override the other requirements, only laptops with the specified hard drive space and RAM would be accepted, no matter their age. The second point i don't have any experience with, so you know better how big of a problem that has been.

I guess students who can afford this program's tuition also can afford a new laptop in most cases, but it would still be nice to be a bit more generous here I think. Maybe I could increase the age span just a little bit to 5 or 6 years, which would be laptops from 2014 or 2015?

File: resources_pages/install_ds_stack_ubuntu.md (hunk 2 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md:22]: Thanks! Opened https://github.com/UBC-MDS/UBC-MDS.github.io/issues/58

File: resources_pages/install_ds_stack_ubuntu.md (hunk 12 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: Good catch, I think just adding surrounding `[]` should work for all hyperlinks in markdown general (without the repeated URL in parenthesis), but not 100% when it comes go GitHub pages. I changed to the same as the line above with a human readable name here.

In general, do you prefer to have the URL visible or hidden behind a human readable name? There is a mix in the installation instructions and I can go through and consolidate, wouldn't take too much time. I think it is nicer if it is hidden, and don't think anyone will be in a situation where they need it printed out since they will need to be connected to the internet to install the software anyways.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 13 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: Agree, updated.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 11 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: Updated

File: resources_pages/install_ds_stack_ubuntu.md (hunk 7 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: Ok, I like when packages use the system update mechanisms, so it unfortunate that there is not a snap or official ppa for rstudio. But not a big deal, I added this note:

> Note that since we installed RStudio directly from a deb file rather than from a repository or a snap package, it will not be updated when we run `sudo apt upgrade` and not automatically as for snap packages. Instead, RStudio will notify you of any available updates when the program is launched.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 14 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: Ok, do you want me to also move back `IRkernel` to it's own section? I thought it would be nice to just have to run `install.packages` once, but I also see the value in having each heading being largely self-contained.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 9 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: Yup works fine, updated the instructions. I also added this brief explanation of what `PATH` is (maybe slightly technically inprecise but helpful at this point):

> (`PATH` is a collection of directories which contain the available executable programs on your computer).

File: resources_pages/install_ds_stack_ubuntu.md (hunk 3 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: Added, let me konw if you think it is too wordy and I can write it elsewhere (maybe it only needs to be on the ubuntu install page?)

> - runs one of the following operating systems: macOS 10.15.X (Catalina), Ubuntu 20.04, Windows 10 Professional, Enterprise or Education; version 2004. *Note that Windows 10 Home is not sufficient as not all the software required for the program can be installed on that OS. Also note that when installing Ubuntu, checking the box "Install third party..." will (among other things) install proprietary drivers, which can be helpful for wifi and graphics cards.*

File: resources_pages/install_ds_stack_ubuntu.md (hunk 6 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md:176]: Hmm, in general I think it s useful to teach the default first, and then what optional flags do. I also have a note below about how to advance in the prompt, but did not include the `yes` otpion, I added it now:

> `conda` will show you the packages that will be downloaded,
> and you can press enter to proceed with the installation.
> If you want to answer `yes` by default and skip this confirmation step, you can replace `conda install` with `conda install -y`.

But you decide, so if you want `-y` by default, I will make the change you suggested and invert the note.

**Review by joelostblom — COMMENTED** [review]
@ttimbers Thanks for the review, see my replies here!

File: resources_pages/install_ds_stack_ubuntu.md (hunk 14 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Yeah maybe... I think conceptually it makes sense so folks know what the individual things are for (tinyTex for LaTeX, IRkernel for the R kernel for Jupyter, and the rest are R packages for other stuff).

File: resources_pages/install_ds_stack_ubuntu.md (hunk 1 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:15]: Can you add this to next weeks academic meeting agenda? This is a bigger issue that I do not feel comfortable for just you and I making the decision to change it.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 3 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Thanks, I will forward this to Vanessa.

Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: We should have this on all the pages to be consistent (and it should match on our marketing website - which I am asking Vanessa to update.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 6 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md:176]: I am happy with this compromise.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 7 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Sounds good!

File: resources_pages/install_ds_stack_ubuntu.md (hunk 9 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Great!

File: resources_pages/install_ds_stack_ubuntu.md (hunk 7 above)
Comment (jennybc) [reply resources_pages/install_ds_stack_ubuntu.md]: Lurker here 🙊

I still think the preview version of RStudio is a good fit for MDS (and most users). The IDE team treats previews almost like release candidates -- once they've started publishing previews, the bar for what new commits become part of that preview release is much higher, and generally they try to offer higher guarantees that things will not break as they iterate on preview releases.

It's possible some of your questions or assessments could be asked or confirmed on https://community.rstudio.com, i.e. best way to install / update on ubuntu. The "R admins" and "RStudio IDE" categories are pretty relevant.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 12 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: I have a slight preference for visible link + hyperlink to make it super obvious its a link they should click on. I sometimes miss a hidden coloured link that you only see because it is coloured. This might be especially problematic for colourblind or visually impaired folks as well...

File: resources_pages/install_ds_stack_ubuntu.md (hunk 13 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Great, thanks!

File: resources_pages/install_ds_stack_ubuntu.md (hunk 7 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Let's listen to @jennybc here and change these instructions (and the Mac and Windows ones as well) to suggest they install the RStudio preview. Thanks for lurking Jenny! It is always helpful to hear your thoughts and wisdom!

**Comment by ttimbers** [comment]
@joelostblom - I have just gone through your review response and made a few more comments. Once you go through these and address them, ping me and I'll call this done and merge it.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 1 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md:15]: Suggested here https://github.ubc.ca/ubc-mds/academic/issues/239#issuecomment-17234

File: resources_pages/install_ds_stack_ubuntu.md (hunk 6 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md:176]: Thanks for compromising!

File: resources_pages/install_ds_stack_ubuntu.md (hunk 7 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: Thanks both! I updated to link the preview version and confirmed that it the Ubuntu 18 version installs well also on 20.04 for the preview.

On the forum I read that the RStudio team is considering a snap/flatpack/appimage release but have not yet decided on which one. I also found out about a [possible issue with high res screens and NVIDIA drivers causing rstudio to crash (only on ubuntu 20.04 it seems)](https://community.rstudio.com/t/rstudio-is-crashing-after-update-on-ubuntu-20-04-focal/66682/13), which could be something for us to keep in mind when troubleshooting students installations.

**Review by joelostblom — COMMENTED** [review]
Should be good to merge now @ttimbers. I will attend your office hours tomorrow also in case we need to discuss something about this quickly before merging.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 12 above)
Comment (joelostblom) [reply resources_pages/install_ds_stack_ubuntu.md]: @ttimbers Agree that only color is not sufficient. I updated to either include the URL as you preferred or mentioned "read more" or similar together with a descriptive name.

For the future, I think a good accessibility practice is using a color of a different contrast/luminosity and underlining links. Regarding text, I believe that short and descriptive is preferred over the actual URL due to increased accessibility for people using screen readers, which read urls letter by letter, and functionalities such as compiling a list of links are not great to listen to (same goes when using only "read more" etc), and navigating by voice is also really hard with long URLs.

Unless you know of someone in this year's cohort that has indicated that they are using a screen reader, maybe we can implement this next year? I opened issue https://github.com/UBC-MDS/UBC-MDS.github.io/issues/59 for this.

Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Sounds good. And no, I am not aware yet of a student who is colour blind or has other visual impairments. I just remember in my ugrad course, a colour blind student was so grateful that we taught the use of colour blind friendly palettes that this has stuck with me. Let's go with this for now but keep that issue open as a reminder to do better next year.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 3 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: This is done now.

File: resources_pages/install_ds_stack_ubuntu.md (hunk 7 above)
Comment (ttimbers) [reply resources_pages/install_ds_stack_ubuntu.md]: Thanks Joel!

**Comment by ttimbers** [comment]
@joelostblom - I think the one thing left to do is to pull out the `IRkernel` package install as discussed above (conceptually this makes more sense even though its a bit more work).

**Comment by joelostblom** [comment]
@ttimbers I did do this in https://github.com/UBC-MDS/UBC-MDS.github.io/pull/56/commits/7aaba85011cfdcd772d8157dba20b50dc9892a13, or are we referring to different things?

**Comment by ttimbers** [comment]
Sorry, that is perfect, I must have clicked on a different commit and saw that it wasn't done. Apologies! Looks good and I will merge now!

**Comment by joelostblom** [comment]
Great! I will start with the win instructions tomorrow, making similar changes as what we agreed upon here.

**Comment by ttimbers** [comment]
Before you move onto that - can you bring the relevant changes we made to ubuntu back to mac?

**Comment by joelostblom** [comment]
Yes, sorry forgot that we said that when I wrote above, I'll do that first tomorrow morning.
